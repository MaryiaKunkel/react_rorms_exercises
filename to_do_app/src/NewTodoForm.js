import React, { useState } from "react";

function NewTodoForm({ addTodo }) {
  const initialState = {
    text: "",
  };
  const [formData, setFormData] = useState(initialState);

  function handleChange(e) {
    const { name, value } = e.target;
    setFormData((data) => ({
      ...data,
      [name]: value,
    }));
  }

  function handleSumbit(e) {
    e.preventDefault();
    addTodo(formData);
    setFormData(initialState);
  }

  return (
    <form onSubmit={handleSumbit}>
      <input
        id="text"
        name="text"
        type="text"
        placeholder="type in your task"
        value={formData.text}
        onChange={handleChange}
      ></input>
      <button>Add task</button>
    </form>
  );
}

export default NewTodoForm;
