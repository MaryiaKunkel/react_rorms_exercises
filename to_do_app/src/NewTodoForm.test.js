import React from "react";
import { render, fireEvent } from "@testing-library/react";
import NewTodoForm from "./NewTodoForm";

it("renders without crashing", function () {
  render(<NewTodoForm />);
});

it("matches snapshot", function () {
  const { asFragment } = render(<NewTodoForm />);
  expect(asFragment()).toMatchSnapshot();
});

it("should update the form data state according to the changes", function () {
  const { getByPlaceholderText } = render(<NewTodoForm />);
  const input = getByPlaceholderText("type in your task");

  fireEvent.change(input, { target: { value: "do dishes" } });

  expect(input.value).toBe("do dishes");
});

it("should call addTodo function when sumbit", function () {
  const addTodoMock = jest.fn();
  const { getByPlaceholderText, getByText } = render(
    <NewTodoForm addTodo={addTodoMock} />
  );
  const input = getByPlaceholderText("type in your task");
  const button = getByText("Add task");

  fireEvent.change(input, { target: { value: "do dishes" } });
  fireEvent.click(button);

  expect(addTodoMock).toHaveBeenCalledWith({
    text: "do dishes",
  });
});
