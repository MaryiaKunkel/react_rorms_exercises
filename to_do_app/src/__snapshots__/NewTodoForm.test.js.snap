// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <input
      id="text"
      name="text"
      placeholder="type in your task"
      type="text"
      value=""
    />
    <button>
      Add task
    </button>
  </form>
</DocumentFragment>
`;
